crates/compiler/mono/src/ir.rs

```diff
@@ -9486,6 +9567,62 @@ impl<'a> Pattern<'a> {
 
         false
     }
+
+    // TODO: vast majority of the time, the patterns will be singleton or empty.
+    // We should introduce a smallvec optimized for the singleton case.
```

Comment:
can we make this return an `impl Iterator` maybe?

Reply:
yeah, I didn't want to implement the iterator - but we can go ahead and do that.